extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampStreamExtractor.java

```diff
@@ -0,0 +1,335 @@
+        return -1;
+    }
+
+    @Nonnull
+    @Override
+    public String getDashMpdUrl() {
+        return "";
+    }
+
+    @Nonnull
+    @Override
+    public String getHlsUrl() {
+        return null;
+    }
+
+    @Override
+    public List<AudioStream> getAudioStreams() {
+        final List<AudioStream> audioStreams = new ArrayList<>();
+
+        audioStreams.add(new AudioStream(
+                albumJson.getArray("trackinfo").getObject(0)
+                        .getObject("file").getString("mp3-128"),
+                MediaFormat.MP3, 128
+        ));
+        return audioStreams;
+    }
+
+    @Override
+    public List<VideoStream> getVideoStreams() {
+        return Collections.EMPTY_LIST;
+    }
+
+    @Override
+    public List<VideoStream> getVideoOnlyStreams() {
+        return Collections.EMPTY_LIST;
+    }
+
+    @Nonnull
+    @Override
+    public List<SubtitlesStream> getSubtitlesDefault() {
+        return new ArrayList<>();
+    }
+
+    @Nonnull
+    @Override
+    public List<SubtitlesStream> getSubtitles(MediaFormat format) {
+        return new ArrayList<>();
+    }
+
+    @Override
+    public StreamType getStreamType() {
+        return StreamType.AUDIO_STREAM;
+    }
+
+    @Override
+    public StreamInfoItemsCollector getRelatedStreams() {
+        return null;
+    }
+
+    @Override
+    public String getErrorMessage() {
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public String getHost() {
+        return "";
+    }
+
+    @Nonnull
+    @Override
+    public String getPrivacy() {
+        return "";
+    }
+
+    @Nonnull
+    @Override
+    public String getCategory() {
+        // Get first tag from html, which is the artist's Genre
+        return document
+                .getElementsByClass("tralbum-tags").first()
+                .getElementsByClass("tag").first().text();
+    }
+
+    @Nonnull
+    @Override
+    public String getLicence() {
+
+        int license = current.getInt("license_type");
+
+        // Tests resulted in this mapping of ints to licence: https://cloud.disroot.org/s/ZTWBxbQ9fKRmRWJ/preview
+
+        switch (license) {
+            case 1:
+                return "All rights reserved ©";
+            case 2:
+                return "CC BY-NC-ND 3.0";
+            case 3:
+                return "CC BY-NC-SA 3.0";
+            case 4:
+                return "CC BY-NC 3.0";
+            case 5:
+                return "CC BY-ND 3.0";
+            case 6:
+                return "CC BY 3.0";
+            case 8:
+                return "CC BY-SA 3.0";
+            default:
+                return "Unknown";
+        }
+    }
+
+    @Nullable
+    @Override
+    public Locale getLanguageInfo() {
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public List<String> getTags() {
+        final Elements tagElements = document.getElementsByAttributeValue("itemprop", "keywords");
+
+        final ArrayList<String> tags = new ArrayList<>();
```

Comment:
`final List<String> tags = new ArrayList<>();`

Work with interface not implementation